Dockerfile

```
@@ -28,6 +28,12 @@ RUN corepack prepare pnpm@latest --activate
 COPY --from=deps /app/node_modules ./node_modules
 COPY . .
 
+# 빌드 시점에 주입될 환경 변수 선언
+ARG BUILD_ENV_VARS
+
+# 전달받은 ARG 환경 변수를 .env.production 파일에 저장
+RUN echo "$BUILD_ENV_VARS" | tr ' ' '\n' > .env.production
```

Hadolint (2.12.0)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)